src/common/catalog-category-registry.ts

```diff
@@ -0,0 +1,56 @@
+import { action, computed, observable, toJS } from "mobx";
+import { CatalogCategory, CatalogEntityData } from "./catalog-entity";
+
+export class CatalogCategoryRegistry {
+  @observable protected categories: CatalogCategory[] = [];
```

Comment:
If you define it as:

```ts
protected categories = observable.array<CatalogCategory>();
```

you can use extended observable-array api to remove item below as (if those are not cloned objects on removal):

```
this.categories.remove(category);
```

Reply:
Good to know but I'm not sure if we have exactly the same instance on removal.